src/Lookup/SiteLinkLookup.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\SiteLink;
+
+/**
+ * Contains methods to lookup of sitelinks of lookup by sitelinks.
+ *
+ * @since 2.0
+ *
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+interface SiteLinkLookup {
+
+	/**
+	 * Returns the id of the item that is equivalent to the
+	 * provided page, or null if there is none.
+	 *
+	 * @since 2.0
+	 *
+	 * @param string $globalSiteId
+	 * @param string $pageTitle
```

Comment:
I am guessing there are cases when we already have the strings and no object.
Slightly quicker to then just pass the strings in.
We also might not have the badges in such cases and thus creating an object with empty / possibly incorrect badges is a bit odd.